test/e2e/common.go

```diff
@@ -212,3 +214,43 @@ func createRestConfig(ctx context.Context, tmpdir, namespace, clusterName string
 
 	return config
 }
+
+// EnsureControlPlaneInitialized waits for the cluster KubeadmControlPlane object to be initialized
+// and then installs cloud-provider-azure components via helm
+// Fulfills the clusterctl.Waiter type so that it can be used as ApplyClusterTemplateAndWaitInput data
+// in the flow of a clusterctl.ApplyClusterTemplateAndWait E2E test scenario
+func EnsureControlPlaneInitialized(ctx context.Context, input clusterctl.ApplyClusterTemplateAndWaitInput, result *clusterctl.ApplyClusterTemplateAndWaitResult) {
+	getter := input.ClusterProxy.GetClient()
+	cluster := framework.GetClusterByName(ctx, framework.GetClusterByNameInput{
+		Getter:    getter,
+		Name:      input.ConfigCluster.ClusterName,
+		Namespace: input.ConfigCluster.Namespace,
+	})
+	kubeadmControlPlane := &kubeadmv1.KubeadmControlPlane{}
+	key := crclient.ObjectKey{
+		Namespace: cluster.Spec.ControlPlaneRef.Namespace,
+		Name:      cluster.Spec.ControlPlaneRef.Name,
+	}
+	Eventually(func() error {
+		return getter.Get(ctx, key, kubeadmControlPlane)
+	}, input.WaitForControlPlaneIntervals...).Should(Succeed(), "Failed to get KubeadmControlPlane object %s/%s", cluster.Spec.ControlPlaneRef.Namespace, cluster.Spec.ControlPlaneRef.Name)
+	if kubeadmControlPlane.Spec.KubeadmConfigSpec.ClusterConfiguration.ControllerManager.ExtraArgs["cluster-cidr"] == "10.244.0.0/16,2001:1234:5678:9a40::/58" {
```

Comment:
@aramase can you think of a general solution to introspect the `cluster-cidr` configuration to detect dual stack in any scenario? would it be to convert the string into a `,`-delimited array, and then to evaluate for len > 1 ?

Reply:
also lemme know if there's a better configuration vector to check than the value of `cluster-cider` for "is a dual stack cluster" or "is an ipv6 cluster"

thanks so much!

Reply:
> @aramase can you think of a general solution to introspect the cluster-cidr configuration to detect dual stack in any scenario? would it be to convert the string into a ,-delimited array, and then to evaluate for len > 1 ?

That's right! We do the similar check in node ipam controller: https://github.com/kubernetes/kubernetes/blob/2a821d787b7f21fbb399b0a0bf6d3fa0c79f4293/cmd/cloud-controller-manager/nodeipamcontroller.go#L179.

Reply:
Do you plan on making the check generic to see if there is an IPv4 and IPv6 CIDR instead of specific check with `"10.244.0.0/16,2001:1234:5678:9a40::/58"`?

Reply:
yep

Reply:
done, thanks for the pro tipz! 🙏